HEP-event-capture/src/models/Form.ts

```diff
@@ -0,0 +1,64 @@
+import { ProgramStage, DataElement } from "../Dhis2Metadata";
+import { CustomForm } from "../components/CustomForm";
+import * as _ from "lodash";
+
+export interface FormData {
+    sections: FormSection[];
+}
+
+export interface FormSection {
+    title: string;
+    dataElements: DataElement[];
+    programStageId: string;
+}
+
+export const formCodes = {
+    formSection: "HEP_POLICY_SECTION",
+    sectionOrder: "SECTION_ORDER",
+    formDataElementGroup: "HEP_POLICY_EVENT_CAPTURE",
+    optionList: "USE_OPTION_FIELD",
+};
+
+export class Form {
+    public static getFormHtml(programStage: ProgramStage): string {
+        const formData = this.getFormData(programStage);
+        const formHtml = CustomForm({ formData });
+        return formHtml;
+    }
+
+    public static getFormData(programStage: ProgramStage): FormData {
+        const allDataElementGroups = programStage.programStageDataElements.map(
+            psde =>
+                psde.dataElement.dataElementGroups.filter(deg =>
+                    new RegExp("^" + formCodes.formSection).test(deg.code)
+                )[0]
+        );
+        const dataElementGroups = _.uniqBy(allDataElementGroups, "code");
+
+        const sections = dataElementGroups.map(deg => {
+            const orderAttribute = deg.attributeValues.find(
+                av => av.attribute.code === formCodes.sectionOrder
+            );
+            const order = orderAttribute && orderAttribute.value;
+            const formattedDataElements = deg.dataElements.map(de => ({
```

Comment:
I understand the simplicity of using the same data structure, but conceptually it would be better to leave dhis2 objects untainted (on the future, they may come from a library where you cannot mess with the interfaces). There are two ways to tackle this: 1) create custom types/interfaces with those custom attributes that extends the base one. 2) Calculate values from the original object. 

I'd do (2) here, by implementing `getUserOptionList(dataElement: DataElement): boolean` somewhere.